src/tracing/esp32_trace/include/matter/tracing/macros_impl.h

```diff
@@ -21,17 +21,100 @@
 #error "Tracing macros seem to be double defined"
 #endif
 
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPError.h>
+
 namespace Insights {
 class ESP32Backend
 {
 public:
     ESP32Backend(const char * str, ...);
     ~ESP32Backend();
 
+    static void AddStringToBlacklist(const char * aLabel)
+    {
+        bool duplicate = false;
+        int i          = 0;
+        for (i = 0; i < blacklistSize; i++)
+        {
+            if (blackListStrings[i] == NULL)
```

Comment:
clang-tidy prefers nullptr instead of NULL. not sure if esp32 code cares though.